server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -418,6 +426,16 @@ private void writeMetadataManifest(String clusterName, String clusterUUID, Clust
         CLUSTER_METADATA_MANIFEST_FORMAT.write(uploadManifest, metadataManifestContainer, fileName, blobStoreRepository.getCompressor());
     }
 
+    private String fetchPreviousClusterUUID(String clusterName, String clusterUUID) {
+        final Optional<ClusterMetadataManifest> latestManifest = getLatestClusterMetadataManifest(clusterName, clusterUUID);
+        if (!latestManifest.isPresent()) {
+            final String previousClusterUUID = getLatestClusterUUID(clusterName);
```

Comment:
getLatestClusterUUID-> getLastKnownUUIDFromRemote
just getLatestClusterUUID is confusing